lib/mix/lib/mix/release.ex

```diff
@@ -333,6 +335,7 @@ defmodule Mix.Release do
     for({app, _mode} <- boot, do: {app, :none})
     |> Keyword.put(:stdlib, :permanent)
     |> Keyword.put(:kernel, :permanent)
+    |> Enum.sort()
```

Comment:
`build_start_clean_boot` uses the output of `build_start_boot`, so there is no need to sort here :)

Reply:
Are you sure? I've added this sort because some tests won't pass (after sorting all apps in test case).

Reply:
Yeah. 

I think the tests are not passing because the apps are not ordered in the `build_release_spec` function. The order to be used is the one specified in `release.boot_scripts.start` (instead of the `Enum.sort(rel_apps)` of the initial implementation of this PR)